platform/viewer/src/routes/Mode/Mode.tsx

```diff
@@ -38,20 +38,41 @@ async function defaultRouteInit({
 
   unsubscriptions.push(instanceAddedUnsubscribe);
 
-  const { unsubscribe: seriesAddedUnsubscribe } = DicomMetadataStore.subscribe(
-    DicomMetadataStore.EVENTS.SERIES_ADDED,
-    ({ StudyInstanceUID, madeInClient }) => {
-      const studyMetadata = DicomMetadataStore.getStudy(StudyInstanceUID);
-      if (!madeInClient) {
-        HangingProtocolService.run(studyMetadata);
-      }
+  const { unsubscribe: seriesAddedUnsubscribe } = DisplaySetService.subscribe(
+    DisplaySetService.EVENTS.DISPLAY_SETS_CHANGED,
+    displaySets => {
+      if (!displaySets || !displaySets.length) return;
+      // The assumption is that the display set at position 0 is the first
+      // study being displayed, and is thus the "active" study.
+      const studyMap = {};
+      const studies = displaySets.
+        reduce((prev, curr) => {
+          const { StudyInstanceUID } = curr;
+          if (!studyMap[StudyInstanceUID]) {
+            const study = DicomMetadataStore.getStudy(StudyInstanceUID);
+            studyMap[StudyInstanceUID] = study;
+            prev.push(study);
+          }
+          return prev;
+        }, []);
+      const studyMetaData = studies[0];
+      HangingProtocolService.run({ studies, studyMetaData, displaySets });
     }
   );
   unsubscriptions.push(seriesAddedUnsubscribe);
 
+  // No point trying to fire partial retrieves, as it causes havoc
+  // when some series are not displayable, and causes the wrong series
+  // to be fetched.
```

Comment:
I'm not sure If I understand why we need to hold off here? Aren't we matching on the displaySets of the current study? Do you need other studies' metadata to load first? Can you please add more comments?

Reply:
Because if the order of the series arrival doesn't match the display set matches, then you never see anything.